filters/openpolicyagent/opaauthorizerequest/opaauthorizerequest.go

```diff
@@ -143,8 +153,34 @@ func addRequestHeaders(fc filters.FilterContext, headers http.Header) {
 		}
 	}
 }
+func (f *opaAuthorizeRequestFilter) Response(fc filters.FilterContext) {
+	span, _ := f.opa.StartSpanFromFilterContext(fc)
```

Comment:
I don't see why we need a custom span here. All response filters are tracked via responseFilters (or ingress) Span anyways and since we have no call to somewhere else, we don't need to have more insights

Reply:
Yes, the whole purpose was to have access to Span as `HandleInvalidDecisionError` method is setting the error details to Span. Modified to use the parent Span for this. Let me know if there is a better way to handle this.